### PR TITLE
Handling historical non-USA audits

### PR DESCRIPTION
**Pull request by phildominguez-gsa**
In this PR:
- This fix addresses some historical audits that have an auditor country of `NON-USA` instead of `non-USA`, which makes them unable to migrate.
- Also accounting for auditor country when checking for empty auditor zipcode
- Fixing/adding unit tests

Testing:
- Find an audit in the elecauditsheader table with `CPACOUNTRY` of "NON-USA", has an empty `CPAZIPCODE`, and non-empty `CPAFOREIGN`, or create one from an existing audit
- Run `historic_data_migrator` on that audit and it should migrate without issue
- The `auditor_country` value in the dissemination_general table should be "non-USA" with an empty `auditor_zip`. `auditor_foreign_address` should match `CPAFOREIGN`.
- If using preview data, dbkey 34158, audit year 2021 should pass

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/census_historical_migration/sac_general_lib/general_information.py [hunk 1]
```diff
@@ -255,6 +255,36 @@ def xform_country(general_information, audit_header):
...
 
 
+def xform_country_v2(general_information, audit_header):
+    """Transforms the country from Census format to FAC format."""
+    auditor_country = string_to_string(
+        general_information.get("auditor_country")
+    ).upper()
+    if auditor_country in ["US", "USA"]:
+        general_information["auditor_country"] = "USA"
+    elif auditor_country == "NON-US":
```
Comment (sambodeme) [inline backend/census_historical_migration/sac_general_lib/general_information.py:265]:
```suggestion
    elif auditor_country == "NON-US" and not (string_to_string(audit_header.CPAZIPCODE) or string_to_string(audit_header.CPASTATE) or string_to_string(audit_header.CPACITY) or string_to_string(audit_header.CPASTREET1)):

```

File: backend/census_historical_migration/test_general_information_xforms.py [hunk 2]
```diff
@@ -462,21 +473,47 @@ def test_auditee_ein_empty_replaced(self):
...
-        """Test that a non-empty auditor_zip and auditee_zip remain unchanged"""
+        """Test that a non-empty US auditor_zip and auditee_zip remain unchanged"""
+        input_data = {
+            "auditee_zip": "10108",
+            "auditor_country": "USA",
+            "auditor_zip": "12345",
+        }
+        self.assertEqual(xform_replace_empty_zips(input_data), input_data)
+
+    def test_non_empty_non_us_auditor_zip(self):
```
Comment (sambodeme) [inline backend/census_historical_migration/test_general_information_xforms.py:512]: This test becomes obsolete based on the previous suggestion.

File: backend/census_historical_migration/sac_general_lib/general_information.py (hunk 1 above)
Comment (phildominguez-gsa) [reply backend/census_historical_migration/sac_general_lib/general_information.py]: Done

File: backend/census_historical_migration/test_general_information_xforms.py (hunk 2 above)
Comment (phildominguez-gsa) [reply backend/census_historical_migration/test_general_information_xforms.py]: Removed

**Pull request merged**